lib/internal/process.js

```diff
@@ -283,6 +283,43 @@ function setupUncaughtExceptionCapture(exceptionHandlerState) {
   };
 }
 
+function setupCompareVersion() {
+  const {
+    majorVersion,
+    minorVersion,
+    patchVersion,
+    prereleaseTag,
+  } = process.release;
+
+  process.release.compareVersion =
+    (major, minor, patch, prerelease = false) => {
+      if (major > majorVersion)
+        return -1;
+      if (major < majorVersion)
+        return 1;
+
+      if (minor > minorVersion)
+        return -1;
+      if (minor < minorVersion)
+        return 1;
+
+      if (prereleaseTag) {
+        if (prereleaseTag === prerelease && patch === patchVersion)
+          return 0;
+        if (patch === patchVersion)
+          return 1;
+      }
+      if (prerelease && patch === patchVersion)
```

Comment:
`patch === patchVersion` can be removed here if this check is moved to the bottom.